akka-http/src/main/scala/akka/http/javadsl/server/directives/WebSocketDirectives.scala

```diff
@@ -68,13 +67,12 @@ abstract class WebSocketDirectives extends SecurityDirectives {
    *
    * To support several subprotocols you may chain several `handleWebSocketMessage` Routes.
    */
-  def handleWebSocketMessagesForOptionalProtocol(handler: Flow[Message, Message, Any], subprotocol: Optional[String]): Route = RouteAdapter {
-    val adapted = scaladsl.Flow[s.Message].map(_.asJava).via(handler).map(_.asScala)
-    D.handleWebSocketMessagesForOptionalProtocol(adapted, subprotocol.asScala)
+  def handleWebSocketMessagesForOptionalProtocol(handler: Flow[Message, Message, NotUsed], subprotocol: Optional[String]): Route = RouteAdapter {
+    D.handleWebSocketMessagesForOptionalProtocol(adapt(handler), subprotocol.asScala)
   }
 
   // TODO this is because scala Message does not extend java Message - we could fix that, but http-core is stable
-  private def adapt[T](handler: Flow[Message, Message, T]): scaladsl.Flow[s.Message, s.Message, NotUsed] = {
+  private def adapt(handler: Flow[Message, Message, NotUsed]): scaladsl.Flow[s.Message, s.Message, NotUsed] = {
```

Comment:
Yes. `Message` in scaladsl  extends Java Message now. However, since we delegate the methods to scaladsl which needs a scala message. We have to cast it to scala message. We can't avoid that. And I think maybe the `TODO` comment is wrong.